### PR TITLE
Remove fully generic circe and replace with semi-auto

### PR DESCRIPTION
**Pull request by rupertbates**
## What are you doing in this PR
This PR removes all uses of circe [fully automatic derivation](https://circe.github.io/circe/codecs/auto-derivation.html#)
and replaces them with [semi-automatic](https://circe.github.io/circe/codecs/semiauto-derivation.html). 

The reasons for this are;
- for consistency as semi-auto is the convention in this repo 
- full auto was causing decoding failures when adding explicit types to implicit encoders and decoders which I was doing to remove warnings from the build

### CODE REVIEW

File: support-frontend/app/admin/settings/Amounts.scala [hunk 1]
```diff
@@ -1,40 +1,52 @@
...
+import com.gu.support.encoding.{Codec, DiscriminatedType}
 
 case class AmountsSelection(
     amounts: List[Int],
     defaultAmount: Int,
     hideChooseYourAmount: Option[Boolean],
 )
 
+object AmountsSelection {
+  implicit val codec: Codec[AmountsSelection] = deriveCodec[AmountsSelection]
```
Comment (johnduffell) [inline support-frontend/app/admin/settings/Amounts.scala:13]: if you prefer you can reduce the repetition https://circe.github.io/circe/codecs/semiauto-derivation.html#:~:text=You%20can%20also,Foo%5D%20%3D%20deriveEncoder
```suggestion
  implicit val codec: Codec[AmountsSelection] = deriveCodec
```

File: support-frontend/app/admin/settings/Switches.scala [hunk 2]
```diff
@@ -22,23 +25,39 @@ case class FeatureSwitches(
     authenticateWithOkta: Option[SwitchState],
 )
 
+object FeatureSwitches {
+  implicit val featureSwitchesCodec = deriveCodec[FeatureSwitches]
```
Comment (johnduffell) [inline support-frontend/app/admin/settings/Switches.scala:29]: missing type annotaion
```suggestion
  implicit val featureSwitchesCodec: Codec[FeatureSwitches] = deriveCodec
```

File: support-frontend/app/admin/settings/Switches.scala [hunk 3]
```diff
@@ -22,23 +25,39 @@ case class FeatureSwitches(
...
+}
+
 case class CampaignSwitches(
     enableContributionsCampaign: Option[SwitchState],
     forceContributionsCampaign: Option[SwitchState],
     usEoy2024: Option[SwitchState] = None,
 )
 
+object CampaignSwitches {
+  implicit val campaignSwitchesCodec = deriveCodec[CampaignSwitches]
```
Comment (johnduffell) [inline support-frontend/app/admin/settings/Switches.scala:39]: type annotation
```suggestion
  implicit val campaignSwitchesCodec: Codec[CampaignSwitches] = deriveCodec
```

File: support-frontend/test/admin/settings/ContributionTypesSpec.scala [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+import scala.io.Source
+
+class ContributionTypesSpec extends AnyWordSpec with Matchers {
+  "ContributionTypes" should {
+    "decode correctly" in {
+      val json: String = Source.fromURL(getClass.getResource("/contributionTypes.json")).mkString
+      val contributionTypes: Either[circe.Error, ContributionTypes] = decode[ContributionTypes](json)
+      contributionTypes.isRight mustBe true
+      contributionTypes.map(contributionTypes => contributionTypes.GBPCountries.size mustBe 3)
```
Comment (johnduffell) [inline support-frontend/test/admin/settings/ContributionTypesSpec.scala:17]: this makes it much nicer
https://www.scalatest.org/user_guide/using_EitherValues
```suggestion
class ContributionTypesSpec extends AnyWordSpec with Matchers with EitherValues {
  "ContributionTypes" should {
    "decode correctly" in {
      val json: String = Source.fromURL(getClass.getResource("/contributionTypes.json")).mkString
      val contributionTypes: Either[circe.Error, ContributionTypes] = decode[ContributionTypes](json)
      contributionTypes.right.value.GBPCountries.size mustBe 3
```

File: support-frontend/test/admin/settings/SettingsIntegrationTest.scala [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      allSettingsProvider <- AllSettingsProvider.fromConfig(configuration)
+      allSettings = allSettingsProvider.getAllSettings()
+    } yield allSettings
+
+    maybeAllSettings match {
+      case Left(e) => fail(e.getMessage)
+      case Right(_) => succeed
+    }
+
+    maybeAllSettings.right.get.amounts.length shouldBe 10
```
Comment (johnduffell) [inline support-frontend/test/admin/settings/SettingsIntegrationTest.scala:42]: add `with EitherValues` to get a nicer error if this fails
```suggestion
    maybeAllSettings.right.value.amounts.length shouldBe 10
```

**Review by johnduffell — APPROVED** [review]
nice - a few small comments but nice to have this all consistent and seconded on the added tests 👏

File: support-frontend/test/admin/settings/SettingsIntegrationTest.scala (hunk 5 above)
Comment (rupertbates) [reply support-frontend/test/admin/settings/SettingsIntegrationTest.scala:42]: I can never remember the name of EitherValues!

**Pull request merged**